react-components/src/hooks/useFetchTimeseriesResourceRelationship.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { type Asset, type Timeseries } from '@cognite/sdk';
+import { isUndefined } from 'lodash';
+import { useEffect } from 'react';
+import { useSDK } from '../components/RevealCanvas/SDKProvider';
+import { getResourceRelationship } from './network/getResourceRelationship';
+
+export const useFetchTimeseriesResourceRelationship = ({
+  asset,
+  filter,
+  callback
+}: {
+  asset: Asset;
+  callback: (data: Timeseries[]) => void;
+  filter?: { isStep?: boolean; isString?: boolean };
+}): void => {
+  const sdk = useSDK();
+
+  useEffect(() => {
+    const getRelationship = async (): Promise<void> => {
+      const relationshipsFound = await getResourceRelationship(
+        sdk,
```

Comment:
It was okay to have this as a async function - but if we want it as a hook, that's fine, but we should use `useQuery` instead of `useEffect` here :+1:

Reply:
moved as useQuery
https://github.com/cognitedata/reveal/pull/4805/commits/cdbaaa721ffd61c0e2916e7d377be59516010121